plugins/processors/emfProcessor/emfProcessor.go

```diff
@@ -40,6 +76,17 @@ func (e *EmfProcessor) Apply(in ...telegraf.Metric) (result []telegraf.Metric) {
 		for _, declaration := range e.MetricDeclarations {
 			declaration.init()
 		}
+
+		// Metric units validation
+		e.validMetricUnit = make(map[string]string)
+		for k, v := range e.MetricUnit {
+			if _, ok := supportedUnits[v]; ok {
+				e.validMetricUnit[k] = v
+			} else {
+				log.Println("E! detect unsupported unit: ", v)
```

Comment:
I think this should be a warning, and Telegraf logging normally would add the plugin name before the log message too. One easy way to do it is follow this pattern and let telegraf take care of the formatting:
https://github.com/influxdata/telegraf/wiki/Logging

Reply:
Thanks Edward for reviewing! 
I changed this logging to be warning, but didn't implement Plugin logging, since currently emfProcessor plugin uses agent logging at other places, and it's a bit stretch for this commit.